### PR TITLE
Updated build pipeline to TS using ESM and Webpack

### PR DESCRIPTION
**Pull request by vim-sroberge**

### CODE REVIEW

**Comment by vim-sroberge** [comment]
Disabled:

- GUI
- Stats

Known Issue: 

- Material Colors

File: VIMLoader.ts [hunk 1]
```diff
@@ -507,17 +489,18 @@ THREE.VIMLoader.prototype =
...
         let r = new THREE.Matrix4();
         r.elements = m;
         return r;
-    },
+    }
 
+    /*
     // Merges all meshes with only a single instance 
-    mergeSingleInstances: function ( instancedMeshList, material )
+    mergeSingleInstances ( instancedMeshList, material )
```
Comment (cdiggins) [inline VIMLoader.ts:503]: Can this be removed?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "vim-webgl-viewer",
-  "version": "0.5.0",
-  "description": "A 3D model viewer in JavaScript.",
+  "version": "1.0.0",
```
Comment (cdiggins) [inline package.json:3]: I wouldn't upgrade to 1.0.0 yet, this is really a 0.6.0 version of the viewer.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "vim-webgl-viewer",
-  "version": "0.5.0",
-  "description": "A 3D model viewer in JavaScript.",
```
Comment (cdiggins) [inline package.json:4]: We will need a description for the package when it is distributed on NPM (https://www.npmjs.com/). Think of NPM as Nuget for JavaScript packages.

File: package.json [hunk 4]
```diff
@@ -15,20 +14,17 @@
     "url": "https://github.com/vimaec/vim-webgl-viewer.git"
   },
   "scripts": {
-    "build": "./node_modules/.bin/gulp -- build",
-    "dist": "./node_modules/.bin/gulp -- dist",
-    "tswatch": "./node_modules/.bin/tsc.cmd -p tsconfig.json --watch",
-    "watch": "./node_modules/.bin/gulp -- watch",
-    "serve": "http-server"
+    "watch": "webpack --watch",
```
Comment (cdiggins) [inline package.json:17]: We are missing a "distribution" step (previously "dist"). The distribution step should produce a minified (release) version of the JavaScript, ready for publishing on NPM (see: https://docs.npmjs.com/cli/v7/commands/npm-publish). We can tackle that separately.

**Pull request merged**